### ISSUE TITLE
[Feature request] Add "boxed" variant to the Testimonial

### ISSUE THREAD

**Issue by seangolob** [primer/brand#829]
Hi PB Team!

I have noticed a few pages that have testimonials that have a boxed wrapper around them:

Single Testimonial:

https://github.com/features/actions
https://github.com/features/issues

![Image](https://github.com/user-attachments/assets/589d9a45-1051-46f4-873d-f17d42fc8511)
![Image](https://github.com/user-attachments/assets/d5542fa3-6f0a-4610-98fc-5f7f6c314d12)

Two Testimonials:

https://github.com/features/codespaces
https://github.com/features/code-search

![Image](https://github.com/user-attachments/assets/297b860a-2b73-4fab-9088-f3ddeb8b9cd3)
![Image](https://github.com/user-attachments/assets/7a061e08-0e04-40a0-b7e6-d7f2bb3e8872)

Can we consider adding a "boxed" variant to the Testimonial component to support these use cases? 

The Single Testimonial page examples have a somewhat transparent background with a blur which should probably exist at the implementation level. However, it would be nice to have the spacing defined in the PB library to have future testimonials share the same spacing.

CC @simmonsjenna CC @jesussandreas

Thank you for your help!

**Comment by rezrah** [comment]
Thanks for opening this feature request @seangolo ❤ 

A couple of options here:

1. Add a Storybook example for your team to reference as the design can be achieved using the `Box` component today
2. Add a `variant` prop to the `Testimonial` component to add first-class support

My preference would be the latter (option 2) assuming Site Design plan to continue using this design approach in future.

There's a _lot_ of detail here that would be difficult to expect users to implement on their own each time. E.g. responsive paddings, responsive filter effects, etc.

I think it's reasonable to add first-class support to Testimonial for this using the variant prop.

E.g. of new API

```typescript
type TesimonialProps {
  variant?: 'default' | 'minimal' | 'frost'
}
```

Like the Card component, 'minimal' would be the correspond to the current version of the Testimonial and be made default to avoid a breaking change. 'default' value would be inclusive of a background color and padded, and frost being the same with a background filter applied.

Final API will likely differ, but I'll bring this to other maintainers and provide an update here when that's available.

**Comment by seangolob** [comment]
@rezrah Thank you for your feedback! ❤ 

Depending on the priority of this issue for the SWP team, we may build this internally and potentially port over to the PB API if/when it becomes available.

For full transparency, this feature request is to support the flex template. In an ideal world, the flex template leverages the built in features from PB and has as few bespoke features built as possible. This should make the template easier to adopt and should make the communication around feature support more clear.

**Comment by rezrah** [comment]
@seangolob - thanks for your patience. Sorry for the delay in responding to you.

Update:

- Testimonial will support the 'boxed' version as an additional `variant` alongside `minimal`
- We'll separately ship a `GlassEffect` component that handles the filter visual effect. 
- Timeline is ~3 weeks for building and shipping these, so you're welcome to find a workaround if you need these sooner.

**Comment by rezrah** [comment]
Update:

- All requested features have shipped in `0.45.0` ([release tracking issue](https://github.com/primer/brand-releases/issues/4))
- ['Boxed' testimonials example here](https://primer.style/brand/components/Testimonial/react#variants)
- Frosted glass effect testimonial: [Light](https://stunning-chainsaw-j82glqz.pages.github.io/?path=/story/components-testimonial-examples--with-frosted-glass) and [Dark](https://stunning-chainsaw-j82glqz.pages.github.io/?path=/story/components-testimonial-examples--with-frosted-glass-dark)

Feel free to reopen if there's anything that was missed.
